x-pack/plugins/aiops/public/cases/register_cases.tsx

```diff
@@ -44,4 +48,28 @@ export function registerChangePointChartsAttachment(
       }),
     }),
   });
+
+  const LogPatternAttachmentComponent = getPatternAnalysisComponent(coreStart, pluginStart);
+
+  cases.attachmentFramework.registerPersistableState({
+    id: CASES_ATTACHMENT_LOG_PATTERN,
+    icon: 'machineLearningApp',
+    displayName: i18n.translate('xpack.aiops.logPatternAnalysis.cases.attachmentName', {
+      defaultMessage: 'Log pattern analysis',
+    }),
+    getAttachmentViewObject: () => ({
```

Comment:
It is optional, but you might also want to include a `getAttachmentRemovalObject` here to define a message for when this attachment is deleted from the case.